efi/preinstall/check_fw_protections_amd64.go

```diff
@@ -114,21 +114,25 @@ func checkPlatformFirmwareProtections(env internal_efi.HostEnvironment, log *tcg
 			// for implementing a D-CRTM with Intel TXT (which I think is SMX).
 			protectedStartupLocalities |= tpm2.LocalityThree | tpm2.LocalityFour
 		}
+		discreteTPM, err = checkIsTpmDiscreteIntel(amd64Env)
```

Comment:
Rather than having this check as part of the platform firmware protection checks, I wonder whether we should move it to the existing `openAndCheckTPM2Device` function. It already has access to the `internal_efi.HostEnvironment` - the downside is that we have to call `determineCPUVendor` twice, but I think that's ok.